File: dcli/template/java/src/main/java/app/Logger.java
package app;

import java.util.Arrays;

/**
 * Simple logger class
 * You can use it to log useful information for the user.
 */
public class Logger {

    {%if targetMethodReturn == "" %}
    public static void logFail() {
        System.out.println("----------------------------------------------------");
        System.err.println("> {{ targetMethod }}() went wrong.");
        System.err.println("Try again.");
    }


    {% else %}
    public static void logFail({{ targetMethodReturn }} expected, {{ targetMethodReturn }} received) {
        System.out.println("----------------------------------------------------");
        System.err.println("> {{ targetMethod }}() = " + received);
        System.err.println("> Expected = " + expected);
        System.err.println("Received value is not correct.");
        System.err.println("Try again.");
    }

    public static void logFail({{ targetMethodReturn }} expected, {{ targetMethodReturn }} received, Object... input) {
        System.out.println("----------------------------------------------------");
        System.err.println("> {{ targetMethod }}(" + Arrays.toString(input) + ") = " + received);
        System.err.println("> Expected = " + expected);
        System.err.println("Received value is not correct.");
        System.err.println("Try again.");
    }
    {% endif %}

    public static void log(String message) {
        System.out.println("----------------------------------------------------");
        System.out.println(message);
        System.out.println("----------------------------------------------------");
    }

    public static void logSuccess() {
        System.out.println("----------------------------------------------------");
        System.out.println("Everything went well!");
        System.out.println("----------------------------------------------------");
    }

    public static void logException(Exception e) {
        System.err.println("----------------------------------------------------");
        System.err.println("An error occurred during runtime.");
        System.err.println("Details:");
        System.err.println(e.getMessage());
    }

    public static void logException(Throwable throwable) {
        System.err.println("----------------------------------------------------");
        System.err.println("Something bad happened!");
        System.err.println(throwable.getMessage());
        throwable.printStackTrace();
        System.err.println("----------------------------------------------------");
    }
}

File: dcli/template/kotlin/src/main/kotlin/app/Logger.kt
package app

import java.util.Arrays

/**
 * Simple logger class
 * You can use it to log useful information for the user.
 */
class Logger {
    companion object {
        {%if targetMethodReturn == "" %}
        fun logFail(): Unit {
            System.out.println("----------------------------------------------------");
            System.err.println("> {{ targetMethod }}() went wrong.");
            System.err.println("Try again.");
        }


        {% else %}
        fun logFail(expected: {{ targetMethodReturn }}, received: {{ targetMethodReturn }}): Unit {
            System.out.println("----------------------------------------------------");
            System.err.println("> {{ targetMethod }}() = " + received);
            System.err.println("> Expected = " + expected);
            System.err.println("Received value is not correct.");
            System.err.println("Try again.");
        }

        fun logFail(expected: {{ targetMethodReturn }}, received: {{ targetMethodReturn }}, vararg input: Any): Unit {
            System.out.println("----------------------------------------------------");
            System.err.println("> {{ targetMethod }}(" + Arrays.toString(input) + ") = " + received);
            System.err.println("> Expected = " + expected);
            System.err.println("Received value is not correct.");
            System.err.println("Try again.");
        }
        {% endif %}

        fun log(message: String): Unit {
            System.out.println("----------------------------------------------------");
            System.out.println(message);
            System.out.println("----------------------------------------------------");
        }

        fun logSuccess(): Unit {
            System.out.println("----------------------------------------------------");
            System.out.println("Everything went well!");
            System.out.println("----------------------------------------------------");
        }

        fun logException(e: Exception): Unit {
            System.err.println("----------------------------------------------------");
            System.err.println("An error occurred during runtime.");
            System.err.println("Details:");
            System.err.println(e.message);
        }

        fun logException(throwable: Throwable): Unit {
            System.err.println("----------------------------------------------------");
            System.err.println("Something bad happened!");
            System.err.println(throwable.message);
            throwable.printStackTrace();
            System.err.println("----------------------------------------------------");
        }
    }
}
